pep_sphinx_extensions/__init__.py

```diff
@@ -37,12 +38,37 @@ def _depart_maths():
     pass  # No-op callable for the type checker
 
 
-def _update_config_for_builder(app: Sphinx):
+def _check_external_builder(app: Sphinx, conf: Config) -> None:
+    if "internal_builder" not in app.tags:  # internal_builder exists if Sphinx is run by build.py
+        conf.html_context = {"external_builder": True}
+    else:
+        conf.html_context = {"external_builder": False}
+    app.tags.remove("internal_builder")
+
+    if conf.html_context["external_builder"]:
+        app.connect("build-finished", _post_build)  # Post-build tasks
+
+
+def _update_config_for_builder(app: Sphinx) -> None:
+    if app.config.html_context["external_builder"]:
+        app.builder.copysource = False  # Prevent unneeded source copying - we link direct to GitHub
+        app.builder.search = False  # Disable search
+
     if app.builder.name == "dirhtml":
         config.pep_url = f"../{config.pep_stem}"
         app.env.settings["pep_file_url_template"] = "../pep-%04d"
 
 
+def _post_build(app: Sphinx, exception: Exception | None) -> None:
+    from pathlib import Path
+
+    from build import create_index_file
```

Comment:
FYI -- this conflicts with https://github.com/pypa/build/ in naming, so... that might be something to be wary of?